src/tools/clippy/clippy_lints/src/transmute/transmute_undefined_repr.rs

```diff
@@ -278,7 +278,7 @@ fn reduce_ty<'tcx>(cx: &LateContext<'tcx>, mut ty: Ty<'tcx>) -> ReducedTy<'tcx>
                     ty = sized_ty;
                     continue;
                 }
-                if def.repr().inhibit_struct_field_reordering_opt() {
+                if def.repr().inhibit_struct_field_reordering() {
```

Comment:
FWIW clippy already used to assume `inhibit_struct_field_reordering_opt` means that the field order is *guaranteed*, which was not the case before this PR.